anndata/_core/index.py

```diff
@@ -71,14 +71,22 @@ def name_idx(i):
     if is_dask(indexer):
         if is_dask(index):
             if isinstance(indexer, dd.Series):
-                indexer = slice(0, len(indexer), 1)
+                indexer = indexer.values
```

Comment:
When handling a regular pd.Series, the normalizer converts to `.values`.  Doing the same here creates an additional dask task.